### ISSUE TITLE
Support to debug (lldb) wasm execution of runtime

### ISSUE THREAD

**Issue by crystalin** [paritytech/polkadot-sdk#4127]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Motivation

The ability to properly debug the wasm runtime would help to solve issues faster

### Request

Having a way to properly build and debug the wasm runtime (with support for VSCode would be additional bonus).

I've tried multiple "debug solutions" already existing, but none are allowing 
- force-debug
- WASM_BUILD_TYPE=debug

Ideally, this should be part of a readme file or described somewhere for people to easily find it :)

### Are you willing to help with this request?

Yes!

**Comment by crystalin** [comment]
I forgot to mention I also tried to directly use "config.debug_info(true)" into the compilation, but that lead to crashes

**Comment by ggwpez** [comment]
> Having a way to properly build and debug the wasm runtime (with support for VSCode would be additional bonus).

Agreed, i think the best we can do so far is to setup a Rust test and use native execution (eg https://github.com/ggwpez/wtfwt). That can at least help with finding issues by stepping through it in a debugger.

But our runtime is still way to debugger-hostile... would be nice if we can step through a WASM call directly.

**Comment by crystalin** [comment]
Agree, we have been using native mostly from the beginning but on one side it is deprecated and on the other we also need to debug the wasm part sometimes

**Comment by crystalin** [comment]
This issue is becoming critical (cc @bkchr ) now that https://github.com/paritytech/polkadot-sdk/pull/4329 is merged

**Comment by ggwpez** [comment]
I think if remote-externalities would support to download the keys on the fly instead of all at once then it could be possible to use the approach from above.
